File: LeetCode/167. Two Sum II - Input Array Is Sorted/Solution.cs
public class Solution
{
    public int[] TwoSum(int[] numbers, int target)
    {
        int l = 0;
        int r = numbers.Length - 1;
        int sum;

        while (l < r)
        {
            sum = numbers[l] + numbers[r];

            if (sum == target) return new int[] { l + 1, r + 1 };
            if (sum > target) r--;
            else l++;
        }

        return new int[] { l, r };
    }
}

File: LeetCode/167. Two Sum II - Input Array Is Sorted/Solution.ts
function twoSum(numbers: number[], target: number): number[] {
    let l : number = 0;
    let r : number = numbers.length - 1;

    while(l < r){
        if(numbers[l] + numbers[r] > target) r--;
        else if(numbers[l] + numbers[r] < target) l++;
        else return [l + 1, r + 1];
    }

    return [l + 1, r + 1];
};